### ISSUE TITLE
Type check of string literal as throws unexpected error in TSX

### ISSUE THREAD

**Issue by liaokaien** [Microsoft/TypeScript#12019]
**TypeScript Version:**  2.0.6 

**Code**

```ts
type GolangTypes = "int64" | "float64" | "bool" | "string" | "slice"

interface Column {
    title: string
    dataIndex: string
    type: GolangTypes 
}

interface ComponentProps extends BaseProps{
    columns: Column[]
}

class TableAnts extends React.Component<ComponentProps, any> {}

// ... some where I render the TableAnts component:

render(){
    const columns = [{
        title: "title",
        dataIndex: "dataIndex",
        type: "int64"
    }]
    return (
        <TableAnts columns={columns}></TableAnts>
    )
}

```

**Expected behavior:**

Since `type` property in `column[0]` is given a value of "int64", it should be compatible with `GolangTypes` defined at `interface Column`.

**Actual behavior:** 
Error from `tsc`: 
```
ERROR in ./js/components/Home.tsx
(46,88): error TS2322: Type '{ title: string; dataIndex: string; type: string; }[]' is not assignable to type 'Column[]'.
  Type '{ title: string; dataIndex: string; type: string; }' is not assignable to type 'Column'.
    Types of property 'type' are incompatible.
      Type 'string' is not assignable to type 'GolangTypes'.
```

**Comment by kitsonk** [comment]
Literals types are not inferred (unless assigned directly to a `const` in 2.1+):

```ts
render(){
    const columns: Columns[] = [{
        title: "title",
        dataIndex: "dataIndex",
        type: "int64"
    }]
    return (
        <TableAnts columns={columns}></TableAnts>
    )
}
```

**Comment by mrapogee** [comment]
Is this behaviour up for discussion? Where is the discussion?
